Source/WebCore/platform/graphics/FontSizeAdjust.h

```diff
@@ -45,6 +46,7 @@ struct FloatMarkableTraits {
 
 struct FontSizeAdjust {
     friend bool operator==(const FontSizeAdjust&, const FontSizeAdjust&) = default;
+    explicit operator bool() const { return value || isFromFont; }
```

Comment:
I would prefer an isNone method than a custom book operator.

Reply:
My intention here is to check if `font-size-adjust` has either a number value or `from-font` with a single method. Otherwise, we need to check two times if FontSizeAdjust has a value, something like `fontSizeAdjust.value || fontSizeAdjust.isFromFont` above. (e.g., FontCascadeCache.h:123,  https://github.com/WebKit/WebKit/pull/18425/files#diff-3dd26ef157ca99d4dd427524bfd9a08c87d6fe42275ad1ed9fde359296067cffL123)

Also, isNone() sounds a bit confusing to me. It could be understood as `font-size-adjust: none`.

Anyhow, I have a plan to clean up the value type here. We can discuss it more in a follow-up patch if you don't mind.

Reply:
I was thinking isNone to represent "is the value none", and then you could check the negation of that to see if from-font or a value was specified.

Reply:
So everywhere you currently check for `fontSizeAdjust`, you would check for `!fontSizeAdjust.isNone()` instead. I think that's nicer and more explicit.

Reply:
I see. But isNone() sounds like a confusing name. What about isEmpty() or hasValue()?

Reply:
Why is it a confusing name? Isn't there a corresponding `font-size-adjust: none` value? My intention is that it maps to that. I'm not a fan of `hasValue()` because technically `from-font` has no value when unresolved. I prefer `isNone` over `isEmpty` for the parallel with the CSS value.

Reply:
Because `font-size-adjust` can have `none` as its value. https://www.w3.org/TR/css-fonts-5/#font-size-adjust-prop

Reply:
Yes, that is my point. isNone would represent when font-size-adjust is set to none

Reply:
@shivamidow To be clear, I think `isNone()` should be defined like: `!fontSizeAdjust.value && !fontSizeAdjust.isFromFont` and then everywhere you currently check for `fontSizeAdjust`, you change it to `!fontSizeAdjust.isNone()`

Reply:
O.K. I will follow up. Stay tuned. Thanks.